src/utils/lm/install.js

```diff
@@ -0,0 +1,262 @@
+
+const getSetupStep = (skipInstall) => {
+  const platform = os.platform()
+  switch (platform) {
+    case 'linux':
+      return setupUnix('Linux', skipInstall)
+    case 'darwin':
+      return setupUnix('Mac OS X', skipInstall)
+    case 'win32':
+      return setupWindows(skipInstall)
+    default:
+      throw new Error(`Platform not supported: ${platform}.
+See manual setup instructions in https://github.com/netlify/netlify-credential-helper#install`)
+  }
+}
+
+const setupGitConfig = async function () {
+  return await configureGitConfig(getHelperPath())
+}
+
+const setupGitConfigStep = {
+  title: `Configuring Git to use Netlify's Git Credential Helper`,
+  task: setupGitConfig,
+}
+
+const installPlatform = async function (force) {
+  const skipInstall = !force && (await installedWithPackageManager())
+  const steps = [
+    checkGitVersionStep,
+    checkGitLFSVersionStep,
+    checkLFSFiltersStep(async (ctx, task, installed) => {
+      if (!installed) {
+        await execa('git', ['lfs', 'install'])
+        task.title += chalk.dim(' [installed]')
+      }
+    }),
+    getSetupStep(skipInstall),
+    setupGitConfigStep,
+  ]
+
+  const tasks = new Listr(steps)
+  await tasks.run()
+
+  return !skipInstall
+}
+
+const skipHelperInstall = function (skip) {
+  if (skip) {
+    return `Netlify's Git Credential Helper already installed with a package manager`
+  }
+}
+
+const installedWithPackageManager = async function () {
+  const installed = hasbin.sync('git-credential-netlify')
+  return installed && !(await fileExistsAsync(getBinPath()))
+}
+
+const setupWindows = function (skipInstall) {
+  return {
+    title: `Installing Netlify's Git Credential Helper for Windows`,
+    skip: () => skipHelperInstall(skipInstall),
+    task: installWithPowershell,
+  }
+}
+
+const installUnixHelper = async function () {
+  const binPath = getBinPath()
+  const shouldFetch = await shouldFetchLatestVersion({
+    binPath,
+    packageName: PACKAGE_NAME,
+    execArgs: ['version'],
+    pattern: `${PACKAGE_NAME}\\/v?([^\\s]+)`,
+    execName: EXEC_NAME,
+  })
+  if (!shouldFetch) {
+    return
+  }
+
+  await fetchLatestVersion({
+    packageName: PACKAGE_NAME,
+    execName: EXEC_NAME,
+    destination: binPath,
+    extension: 'tar.gz',
+  })
+}
+
+const setupUnix = function (platformName, skipInstall) {
+  return {
+    title: `Installing Netlify's Git Credential Helper for ${platformName}`,
+    skip: () => skipHelperInstall(skipInstall),
+    task: async () => {
+      await installUnixHelper()
+      await setupUnixPath()
+    },
+  }
+}
+
+const installWithPowershell = async function () {
+  const script = `[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+iex (iwr -UseBasicParsing -Uri https://raw.githubusercontent.com/netlify/netlify-credential-helper/53fbf6e2b82d1ae366d096a24022fa259d19329a/resources/install.ps1)`
```

Comment:
Should we inline that file in this repository instead of fetching it over the network?

Reply:
Looking at the file:
https://github.com/netlify/netlify-credential-helper/blob/53fbf6e2b82d1ae366d096a24022fa259d19329a/resources/install.ps1
Looks like it would be better to port most of it to JavaScript (at least fetching the release part).

Reply:
Yes, that'd be even better.

Not relying on network requests would also be a win in terms of performance and reliability.

Reply:
This is done in https://github.com/netlify/cli/pull/1489/commits/f693f79abbb3d0d38313f435ffdebcd542800c17